### ISSUE TITLE
`WRONG_OVERLOADING_FUNCTION_ARGUMENTS` false positive with mixed inline/non-inline overloads

### ISSUE THREAD

**Issue by 0x6675636b796f75676974687562** [saveourtool/diktat#1593]
Consider the following fragment:

```kotlin
inline fun <reified T> getLogger(): Logger = getLogger(T::class)

fun getLogger(clazz: Class<*>): Logger = LoggerFactory.getLogger(clazz)

fun getLogger(clazz: KClass<*>): Logger = getLogger(clazz.java)
```

_Diktat_ reports a `WRONG_OVERLOADING_FUNCTION_ARGUMENTS` false positive, despite:

1. the 1st function is an `inline` one, and
2. the 1st function has a _type parameter_ while others do not (so, technically, the 1st function may also be considered having **exactly one** argument)

## Environment information
* diktat version: **1.2.4.2**

## Related

 * #1526
 * #1594

**Comment by icemachined** [comment]
the issue is not reproduced in diktat version 1.2.4.2, it is reproduced in 1.2.4.1 which was cut before #1594 was merged

**Comment by 0x6675636b796f75676974687562** [comment]
Duplicate of #1594
